examples/chat/chat.py

```diff
@@ -87,25 +89,38 @@ def main() -> None:
         "/ip4/127.0.0.1/tcp/8000/p2p/QmQn4SwGkDZKkUEpBRBvTmheQycxAHJUNmVEnjA2v1qe8Q"
     )
     parser = argparse.ArgumentParser(description=description)
-    parser.add_argument(
-        "-p", "--port", default=8000, type=int, help="source port number"
-    )
+    parser.add_argument("-p", "--port", type=int, help="source port number (optional)")
     parser.add_argument(
         "-d",
         "--destination",
         type=str,
         help=f"destination multiaddr string, e.g. {example_maddr}",
     )
     args = parser.parse_args()
-
-    if not args.port:
-        raise RuntimeError("was not able to determine a local port")
+    port = args.port if args.port is not None else get_random_available_port()
 
     try:
-        trio.run(run, *(args.port, args.destination))
+        trio.run(run, *(port, args.destination))
     except KeyboardInterrupt:
         pass
 
 
+def is_port_available(port: int) -> bool:
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+        try:
+            sock.bind(("", port))
+            return True
+        except OSError:
+            return False
+
+
+def get_random_available_port(start: int = 10000, end: int = 20000) -> int:
+    for _ in range(50):  # try up to 50 random ports
+        port = random.randint(start, end)
+        if is_port_available(port):
+            return port
+    raise RuntimeError("Could not find an available port in the given range.")
```

Comment:
@lla-dane  An easy to solve this to simply use default port as 0, and then the application will automatically bind to a available port.

Reply:
The address displayed here is incorrect, as it is generated from the following string:
`f"-d /ip4/{localhost_ip}/tcp/{port}/p2p/{host.get_id().pretty()}\n"`

If you replace this with something like
`f"-d {host.get_addrs()[0]}\n"`
it would print the actual address and the port it is bind to